src/colvarvalue.cpp

```diff
@@ -600,25 +668,30 @@ colvarvalue colvarvalue::dist2_grad(colvarvalue const &x2) const
     return 2.0 * (this->real_value - x2.real_value);
   case colvarvalue::type_3vector:
     return 2.0 * (this->rvector_value - x2.rvector_value);
-  case colvarvalue::type_unit3vector:
-  case colvarvalue::type_unit3vectorderiv:
-    {
-      cvm::rvector const &v1 = this->rvector_value;
-      cvm::rvector const &v2 = x2.rvector_value;
-      cvm::real const cos_t = v1 * v2;
-      return colvarvalue(2.0 * (cos_t * v1 - v2), colvarvalue::type_unit3vectorderiv);
-    }
+  case colvarvalue::type_unit3vector: {
+    cvm::rvector const &v1 = this->rvector_value;
+    cvm::rvector const &v2 = x2.rvector_value;
+    cvm::real const cos_t = v1 * v2;
+    return colvarvalue(2.0 * std::acos(cos_t) * -1.0 / cvm::sqrt(1.0 - cos_t * cos_t) * v2,
```

Comment:
Actually I am still not quite sure about the meaning of `dist2_grad`. For a call like `x1.dist2_grad(x2)`, does it calculating the partial derivative of distance with respect to $\mathbf{x}_1$ or $\mathbf{x}_2$?

Reply:
The former (the scalar version is 2 * (x1 - x2), which for x2 == 0 has the same sign as x1).

Admittedly this could be clarified by adding `colvarvalue::dist2_lgrad()` and `colvarvalue::dist2_rgrad()`. Or rather removing the latter and naming `dist2_grad` to `gradient_distance_from()`? Either way not for this PR.

Reply:
I don't think a code change is necessary. You could add some comments before `dist2_grad` to help people figure out whether it is $\partial d^2 / \partial x_2$ or $\partial d^2 / \partial x_1$.

Reply:
A comment was already near the declaration (`colvarvalue.h`), I just added another one in the definition (`colvarvalue.cpp`) as a commit straight to `master.